### PR TITLE
[adaptive_scaffold] New and Improved Helper Widgets & Updated Examples and Their Tests

### PR DESCRIPTION
**Pull request by gspencergoog**
This builds on https://github.com/flutter/packages/pull/2406, and is the work of @rena987, cloned from https://github.com/flutter/packages/pull/2439, I'm just landing it for her, since she has left.

Added builders and logic to the existing merged pull request above and updated examples and their respective tests.

### CODE REVIEW

File: packages/adaptive_scaffold/example/lib/main.dart [hunk 1]
```diff
@@ -379,46 +469,51 @@ class _ItemList extends StatelessWidget {
...
+                  border: OutlineInputBorder(
+                    borderRadius: BorderRadius.circular(30.0),
+                    borderSide: BorderSide.none,
+                  ),
+                  filled: true,
+                  contentPadding: const EdgeInsets.all(25),
+                  hintStyle: const TextStyle(
+                      color: Color.fromARGB(255, 135, 129, 138)),
+                  hintText: 'Search replies',
+                  fillColor: Colors.white),
```
Comment (dkwingsmt) [inline packages/adaptive_scaffold/example/lib/main.dart:504]:
```suggestion
                  fillColor: Colors.white,
                ),
```

File: packages/adaptive_scaffold/example/lib/main.dart [hunk 2]
```diff
@@ -379,46 +469,51 @@ class _ItemList extends StatelessWidget {
...
-              itemBuilder: (_, int index) => _ItemListTile(
-                selected: selected,
-                item: items[index],
-                selectCard: selectCard,
-              ),
+              itemBuilder: (BuildContext context, int index) => _ItemListTile(
+                  item: items[index],
+                  email: items[index].emails![0],
+                  selectCard: selectCard,
+                  selected: selected),
```
Comment (dkwingsmt) [inline packages/adaptive_scaffold/example/lib/main.dart:514]:
```suggestion
                item: items[index],
                email: items[index].emails![0],
                selectCard: selectCard,
                selected: selected,
              ),
```

File: packages/adaptive_scaffold/example/lib/main.dart [hunk 3]
```diff
@@ -379,46 +469,51 @@ class _ItemList extends StatelessWidget {
...
             ),
-          ),
+          )
```
Comment (dkwingsmt) [inline packages/adaptive_scaffold/example/lib/main.dart:516]:
```suggestion
          ),
```

File: packages/adaptive_scaffold/example/lib/main.dart [hunk 4]
```diff
@@ -521,25 +633,221 @@ class _DetailTile extends StatelessWidget {
...
+                        recepients: thisEmail.recepients,
+                        body: thisEmail.body,
+                        bodyImage: thisEmail.bodyImage);
+                  },
                 ),
-                Text(item.body, style: Theme.of(context).textTheme.bodyLarge),
-              ],
-            ),
+              ),
+              //Text(item.body, style: Theme.of(context).textTheme.bodyLarge),
```
Comment (dkwingsmt) [inline packages/adaptive_scaffold/example/lib/main.dart:715]:
```suggestion
```

**Review by dkwingsmt — APPROVED** [review]
A few nit format suggestions

**Comment by gspencergoog** [comment]
@stuartmorgan Could you please explain to the CLA bot that @rena987 did all the work while a Google intern? I've asked Serena to accept the CLA, but I'm not sure when/if she'll have time to do that, so it would expedite getting this committed to let the bot know that it's OK.

**Comment by stuartmorgan** [comment]
> @stuartmorgan Could you please explain to the CLA bot that @rena987 did all the work while a Google intern?

I don't have the ability to submit an override; I think you need a repo admit like @Hixie.

**Pull request merged**